src/components/TodoItem.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+    <li>
+        <div v-if="!isEditing">
+            {{ todo.message }}
+            <button id="edit" @click="edit">Edit</button>
+            <button id="delete" @click="delToDo">Delete</button>
+        </div>
+        <div v-else>
+            <input v-model="temp" />
+            <button id="save" @click="save">Save</button>
+            <button @click="cancel">Cancel</button>
+        </div>
+    </li>
+</template>
+
+<script>
+    export default {
+        name: "TodoItem",
+        props: ["todo", "index"],
+        data() {
+            return {
+                isEditing: this.todo.isEditing,
+                temp: ""
+            };
+        },
+        methods: {
+            edit: function() {
+                this.temp = this.todo.message;
+                this.isEditing = true;
+            },
+            delToDo: function() {
+                this.isEditing = false;
+                this.$emit("deleteEntry", this.index);
+                // this.$parent.deleteEntry(this.index);
```

Comment:
Sry I didn't  notice it. 
Is it better to use `this.$parent.deleteEntry` or `this.$emit("deleteEntry"` in this particular case?

Reply:
I didn't know `this.$parent` searched the internet and found https://vuejs.org/v2/guide/components-edge-cases.html

I saw that the title is "handling edge cases" and that's enough for me to say: You should really `this.$emit` here, because the parent component can decide on its own what to do with the event.

`this.$parent.doSomething` hard-wires the child component with the parent component, because that way it can only live inside a parent component that has a method `deleteEntry`.

Reply:
ok thank you.
We changed it back to using `$emit` already.